noir-projects/aztec-nr/aztec/src/event/event_interface.nr

```diff
@@ -14,50 +14,136 @@ use dep::protocol_types::{
+/// assumptions.
+///
+/// We are currently still exploring the design space but it seems that there are only 2 meaningful
+/// combinations:
+///
+/// +--------------------------------------------------------------------------------------------------------+
+/// | 1. Constrained encryption, constrained tagging, private log stream                                     |
+/// |                                                                                                        |
+/// | This combination gives us an on-chain guarantee that a recipient is able to discover, receive and      |
+/// | decrypt a message. This is to be used when a smart contract (rather than a person or some offchain     |
+/// | system) needs to make decisions based on the message. For example, consider a contract that escrows a  |
+/// | privately-stored NFT (i.e. an NFT represented by a note) and releases it to a buyer only after         |
+/// | receiving a payment in a specific token. Without on-chain delivery and guaranteed discoverability      |
+/// | afforded by constrained tagging and encryption, the buyer could potentially obtain the NFT without     |
+/// | sending a valid payment message (i.e. the token note hash preimage) to the seller, rugging the seller. |
+/// |                                                                                                        |
+/// | (To clarify the above, while the malicious buyer's payment token would still be deducted from their    |
+/// | balance, they would obtain the NFT while the seller would be unable to spend the payment token,        |
+/// | keeping the payment token note in limbo.)                                                              |
+/// +--------------------------------------------------------------------------------------------------------+
+///
+/// +--------------------------------------------------------------------------------------------------------+
+/// | 2. Unconstrained encryption, offchain message                                                          |
+/// |                                                                                                        |
+/// | This combination is useful when we either trust the sender to behave or the sender is incentivized to  |
+/// | deliver the message. To give you an example let's consider a payment app. When you are buying          |
+/// | groceries, the merchant can consider the payment successful once he receives the payment message, can  |
+/// | successfully decrypt it and see that its contents correspond to a valid token note.                    |
+/// +--------------------------------------------------------------------------------------------------------+
+///
+///
+/// Other combinations like unconstrained encryption, unconstrained tagging, private log stream don't seem
+/// to be meaningful because they are costlier than combination 2 above while providing no meaningful
+/// additional guarantees.
```

Comment:
Unconstrained onchain delivery is very much relevant. it means you don't need any third party solution to handle the delivery, you don't need servers, an out of band channel, backups, nothing. I can just send you tokens from my wallet and it'll work with zero risk. It is very valid.